meetings/working-groups/field-keyword/FK-2024-08-07.md

```diff
@@ -0,0 +1,125 @@
+
+### `[AllowNull]` nullability hole
+
+The proposal mentions that there is an existing nullability hole in the language when `[AllowNull]` is applied to a field, and that this hole will be inherited by `[field: AllowNull]`:
+
+```cs
+// C# 12
+class C
+{
+    [AllowNull]
+    private string resetIfSetToDefault;
+
+    public string ResetIfSetToDefault
+    {
+        // Removed `?? GetDefault()` without warnings!
+        get => resetIfSetToDefault; 
+        set => resetIfSetToDefault = value == GetDefault() ? null : value;
+    }
+}
+```
+
+The point was raised that when it comes to `[field: AllowNull]`, we might have an opportunity to close that gap in warnings because the backing field is scoped to the property. For instance, perhaps we could simply create a nullability warning if `[field: AllowNull]` is used without `[field: MaybeNull]`.
+
+As mentioned elsewhere, `[field: AllowNull]` is likely only going to be specified in advanced scenarios where automatic backing field nullability (somewhat equivalent to AllowNull and MaybeNull together) isn't what the user wants.
+
+### Open question: unrelated nullability warnings
+
+We considered whether unrelated nullability warnings should affect the nullability of the backing field:
+
+```cs
+// Is this a null-resilient getter?
+get
+{
+    string unrelated = null; // Warning
+    return field ?? "";
+}
+```
+
+It would likely not be expensive to detect relatedness. It would consist of doing two nullability analysis passes, only considering warnings that were present in a maybe-null-field pass and not present in a non-null-field pass. There could be occasional scenarios with inferred generic type arguments where there is a different error in each pass; however, what would fall out in that case would be a "not null-resilient" detection, which is a safe place to land, still closer to the mark than doing only a single pass. Finally, it might be harder to explain and understand "are there new warnings caused by `field` becoming nullable" compared to understanding "are there any warnings when `field` is nullable."
+
+We decided to implement only one pass for now and see how it does in practice during implementation. We can revisit this if it seems that the single pass is leaving something to be desired.
+
+### Open question: interaction with manually-applied attributes
+
+We decided that if any nullability attribute is applied using a `field:` target, including the preconditions AllowNull and DisallowNull, then null resilience will not be considered because the user has already entered the space of taking control over the nullability of the backing field. Someone specifying a nullability attribute is likely using it as an escape hatch because the null resilience check wasn't what they wanted.
```

Comment:
```suggestion
We decided that if any nullability attribute is applied using a `field:` target, including the preconditions AllowNull and DisallowNull, then null resilience will not be considered. The user has already entered the space of taking control over the nullability of the backing field. Someone specifying a nullability attribute is likely using it as an escape hatch because the null resilience check wasn't what they wanted.
```